packages/compiler/src/server/server.ts

```diff
@@ -106,13 +107,24 @@ function main() {
     return await s.initialize(params);
   });
 
-  connection.onInitialized((params) => {
+  connection.onInitialized(async (params) => {
     if (clientHasWorkspaceFolderCapability) {
       connection.workspace.onDidChangeWorkspaceFolders(s.workspaceFoldersChanged);
     }
+    // Initialize vscode settings
+    await initializeVsCodeSettings(connection, host);
```

Comment:
let's make the client-config an instance that's passed to compiler-service when creating compiler-service. Maybe call it clientConfigProvider. and you can call it's create method to create a instance to pass to compiler-service.